src/test/java/reposense/parser/ArgsParserTest.java

```diff
@@ -0,0 +1,175 @@
+package reposense.parser;
+
+import java.io.File;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Calendar;
+import java.util.Date;
+import java.util.Optional;
+
+import org.junit.Assert;
+import org.junit.Test;
+
+import reposense.exception.ParseException;
+import reposense.frontend.CliArguments;
+
+public class ArgsParserTest {
+
+    private static final String INPUT_DELIMITER = " ";
+    private static final String PROJECT_DIRECTORY = System.getProperty("user.dir");
+
+    private static final String CONFIG_FILE_ABSOLUTE = Paths.get(PROJECT_DIRECTORY, "sample_small.csv").toString();
+    private static final String OUTPUT_DIRECTORY_ABSOLUTE = Paths.get(PROJECT_DIRECTORY, "build").toString();
+
+    private static final String CONFIG_FILE_RELATIVE = Paths.get("sample_small.csv").toString();
+    private static final String OUTPUT_DIRECTORY_RELATIVE = Paths.get("build").toString();
+
+    private Date getDate(int year, int month, int date) {
+        return new Calendar
+                .Builder()
+                .setDate(year, month, date)
+                .setTimeOfDay(0, 0, 0)
+                .build()
+                .getTime();
+    }
+
+    @Test
+    public void parse_allCorrectInputs_success() throws ParseException {
+        String input = String.format("-config %s -output %s -since 01/07/2017 -until 30/11/2017",
+                CONFIG_FILE_ABSOLUTE, OUTPUT_DIRECTORY_ABSOLUTE);
+        CliArguments cliArguments = ArgsParser.parse(input.split(INPUT_DELIMITER));
+        Assert.assertEquals(CONFIG_FILE_ABSOLUTE, cliArguments.getConfigFilePath().toString());
+        Assert.assertEquals(OUTPUT_DIRECTORY_ABSOLUTE, cliArguments.getTargetFilePath().toString());
+
+        Date expectedSinceDate = getDate(2017, Calendar.JULY, 1);
+        Date expectedUntilDate = getDate(2017, Calendar.NOVEMBER, 30);
+        Assert.assertEquals(expectedSinceDate, cliArguments.getSinceDate().get());
+        Assert.assertEquals(expectedUntilDate, cliArguments.getUntilDate().get());
+    }
+
+    @Test
+    public void parse_configFileOnly_success() throws ParseException {
+        String input = String.format("-config %s", CONFIG_FILE_ABSOLUTE);
+        CliArguments cliArgumentsOfAbsolutePath =  ArgsParser.parse(input.split(INPUT_DELIMITER));
+
+        input = String.format("-config %s", CONFIG_FILE_RELATIVE);
+        CliArguments cliArgumentsOfRelativePath = ArgsParser.parse(input.split(INPUT_DELIMITER));
+
+        Path expected = Paths.get(CONFIG_FILE_ABSOLUTE);
+        Assert.assertEquals(expected, cliArgumentsOfAbsolutePath.getConfigFilePath());
+        Assert.assertEquals(expected, cliArgumentsOfRelativePath.getConfigFilePath());
+    }
+
+    @Test
+    public void parse_configFileAndOutputDirectory_success() throws ParseException {
+        String input = String.format("-config %s -output %s", CONFIG_FILE_ABSOLUTE, OUTPUT_DIRECTORY_RELATIVE);
```

Comment:
for all these tests and below, can we have a variable:
`private static final String DEFAULT_MANDATORY_ARGS = "-config " + CONFIG_FILE_ABSOLUTE;"`
then each of the other tests will use this `DEFAULT_MANDATORY_ARGS + String.format(...))`, make it easier to see which specific argument we testing

Reply:
Done. :+1: